backend/src/users.rs

```diff
@@ -0,0 +1,179 @@
+        fairing::Info {
+            name: "Add CORS headers to responses",
+            kind: fairing::Kind::Response,
+        }
+    }
+
+    async fn on_response<'r>(&self, request: &'r Request<'_>, response: &mut Response<'r>) {
+        response.set_header(http::Header::new(
+            "Access-Control-Allow-Origin",
+            request.headers().get_one("Origin").unwrap_or("*"),
+        ));
+        response.set_header(http::Header::new(
+            "Access-Control-Allow-Methods",
+            "POST, GET, PATCH, OPTIONS",
+        ));
+        response.set_header(http::Header::new("Access-Control-Allow-Headers", "*"));
+        response.set_header(http::Header::new(
+            "Access-Control-Allow-Credentials",
+            "true",
+        ));
+    }
+}
+
+#[get("/users")]
+pub async fn list_users(mut db: Connection<FlukeDb>) -> Result<Json<Vec<UserModel>>> {
+    let list_of_users: Vec<UserModel> = sqlx::query_as!(UserModel, "SELECT * FROM user_profile")
+        .fetch_all(&mut *db)
+        .await?;
+
+    Ok(Json(list_of_users))
+}
+
+pub async fn create_user(
+    mut db: Connection<FlukeDb>,
+    user: CreateUserSchema,
+) -> Result<UserModel, SignupError> {
+    let user_model: UserModel = sqlx::query_as!(
+        UserModel,
+        r#"
+        INSERT INTO user_profile (username, first_name, last_name, email, password)
+        VALUES ($1, $2, $3, $4, $5)
+        RETURNING *
+        "#,
+        user.username,
+        user.first_name,
+        user.last_name,
+        user.email,
+        user.password
+    )
+    .fetch_one(&mut *db)
+    .await
+    .map_err(SignupError::from)?;
+
+    Ok(user_model)
+}
+
+#[post("/signup", data = "<user>")]
+async fn signup_user(
+    db: Connection<FlukeDb>,
+    user: Json<CreateUserSchema>,
+) -> Result<Created<Json<UserModel>>, rocket::response::status::Custom<String>> {
+    match create_user(db, user.into_inner()).await {
+        Ok(user_model) => {
+            let location = format!("/users/{}", user_model.id);
+            Ok(Created::new(location).body(Json(user_model)))
+        }
+        Err(e) => {
+            let status = match e {
+                SignupError::NonUniqueIdError => Status::Conflict,
+                _ => Status::InternalServerError,
+            };
+            Err(rocket::response::status::Custom(status, e.to_string()))
+        }
+    }
+}
+
+#[delete("/users/<id>")]
+pub async fn delete_user(mut db: Connection<FlukeDb>, id: i64) -> Result<Option<()>> {
+    let result: sqlx::postgres::PgQueryResult =
+        sqlx::query!("DELETE FROM user_profile WHERE id = $1", id)
+            .execute(&mut *db)
+            .await?;
+
+    Ok((result.rows_affected() == 1).then(|| ()))
+}
+
+pub fn users_stage() -> AdHoc {
+    AdHoc::on_ignite("Users Stage", |rocket| async {
+        rocket
+            .mount("/users/", routes![list_users, delete_user])
+            .mount("/", routes![signup_user])
+            .attach(CORS)
```

Comment:
can do this piece in the `main` config of rocket server

Reply:
4aa22fd07d12781bb6801b7898d735f648de7193